src/lib/mdns/Advertiser.h

```diff
@@ -51,11 +52,37 @@ class BaseAdvertisingParams
         return *reinterpret_cast<Derived *>(this);
     }
     bool IsIPv4Enabled() const { return mEnableIPv4; }
+    Derived & SetMac(uint8_t * mac, size_t size)
```

Comment:
nit: consider using chip::Span to represent a buffer+length pair

Reply:
Added in next commit - thanks. Still finding all the chip goodies in the codebase :)